From d048b0ee0272eac78f25cc8ba9535efa795d8bad Mon Sep 17 00:00:00 2001
From: =?utf8?q?Christian=20Gim=C3=A9nez?= <cnngimenez@gmail.com>
Date: Tue, 8 Jul 2014 19:00:52 -0300
Subject: [PATCH 1/1] For libpng 1.5.13 the jmpbuf pointer for the png_structp
 doesn't exists, we need to use png_jmpbuf() function. See libpng-manual.txt
 at the /usr/share/doc/libpng-xxxx directory.

---
 png.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git png.c png.c
index 2b59269..52b6797 100644
--- a/png.c
+++ b/png.c
@@ -69,7 +69,7 @@ int fh_png_load(char *name,unsigned char *buffer, unsigned char ** alpha,int x,i
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
@@ -161,7 +161,7 @@ int fh_png_getsize(char *name,int *x,int *y)
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
-- 
2.4.10.GIT

